packages/bot-skeleton/src/services/api/api-base.js

```diff
@@ -0,0 +1,117 @@
+import { observer as globalObserver } from '../../utils/observer';
+import { generateDerivApiInstance, getLoginId, getToken } from './appId';
+
+class APIBase {
+    api;
+    token;
+    account_id;
+    pip_sizes = {};
+    account_info = {};
+    is_running = false;
+    subscriptions = [];
+    time_interval = null;
+
+    init(force_update = false) {
+        if (getLoginId()) {
+            this.toggleRunButton(true);
+            if (force_update && this.api) this.api.disconnect();
+            this.api = generateDerivApiInstance();
+            this.initEventListeners();
+            this.authorizeAndSubscribe();
+            if (this.time_interval) clearInterval(this.time_interval);
+            this.time_interval = null;
+            this.getTime();
+        }
+    }
+
+    initEventListeners() {
+        if (window) {
+            window.addEventListener('online', this.reconnectIfNotConnected);
+            window.addEventListener('focus', this.reconnectIfNotConnected);
+        }
+    }
+
+    createNewInstance(account_id) {
+        if (this.account_id !== account_id) {
+            this.init(true);
+        }
+    }
+
+    reconnectIfNotConnected = () => {
+        // eslint-disable-next-line no-console
+        console.log('connection state: ', this.api.connection.readyState);
+        if (this.api.connection.readyState !== 1) {
+            // eslint-disable-next-line no-console
+            console.log('Info: Connection to the server was closed, trying to reconnect.');
```

Comment:
**suggestion:** could we instead display the error message within the page rather than in the console? perhaps a small toastbox/banner/modal

Reply:
We do not want to show anything to the user, because under the hood the connection will be re initialised, this console is only for dev purpose and will be removed in further releases. @ameerul-deriv